internal/cmd/kafka/command_cluster_create.go

```diff
@@ -142,32 +123,48 @@ func (c *clusterCommand) create(cmd *cobra.Command, args []string, prompt form.P
 		return err
 	}
 
-	encryptionKey, err := cmd.Flags().GetString("encryption-key")
+	var encryptionPtr *string
+	if cmd.Flags().Changed("encryption-key") {
+		if cloud != "gcp" {
+			return errors.New(errors.EncryptionKeySupportErrorMsg)
+		}
+
+		encryptionKey, err := cmd.Flags().GetString("encryption-key")
+		if err != nil {
+			return err
+		}
+		encryptionPtr = &encryptionKey
+
+		if err := c.validateGCPEncryptionKey(cmd, prompt, cloud, c.EnvironmentId()); err != nil {
+			return err
+		}
+	}
+
+	byok, err := cmd.Flags().GetString("byok")
 	if err != nil {
 		return err
 	}
 
-	if encryptionKey != "" {
-		input := validateEncryptionKeyInput{
-			Cloud:          cloud,
-			MetadataClouds: clouds,
-			AccountID:      c.EnvironmentId(),
+	var keyGlobalObjectReference *cmkv2.GlobalObjectReference
+	if byok != "" {
+		key, httpResp, err := c.V2Client.GetByokKey(byok)
+		if err != nil {
+			return errors.CatchByokKeyNotFoundError(err, httpResp)
 		}
-		if err := c.validateEncryptionKey(cmd, prompt, input); err != nil {
-			return err
+		keyGlobalObjectReference = &cmkv2.GlobalObjectReference{
+			Id: key.GetId(),
 		}
```

Comment:
It's fine to move this to one line if there's only one field.
```suggestion
		keyGlobalObjectReference = &cmkv2.GlobalObjectReference{Id: key.GetId()}
```

Reply:
Fixed.